balboa-agent/src/main/scala/com/socrata/balboa/agent/BalboaAgent.scala

```diff
@@ -105,15 +107,30 @@ object BalboaAgent extends App with Config with StrictLogging {
 
   val future: ScheduledFuture[_] = scheduler.scheduleWithFixedDelay(new Runnable {
     override def run(): Unit = {
-      val mc = new MetricConsumer(dataDir, new MetricJmsQueueNotSingleton(amqConnection, activemqQueue))
       try {
-        logger debug s"Attempting to run Metric Consumer $mc"
-        mc.run() // Recursively reads all metric data files and writes them to a queue.
+        val mc = new MetricConsumer(dataDir, new MetricJmsQueueNotSingleton(amqConnection, activemqQueue))
+        try {
+          logger debug s"Attempting to run Metric Consumer $mc"
+          mc.run() // Recursively reads all metric data files and writes them to a queue.
+        } catch {
+          case e: Exception => logger.error("Metric consumer failed", e)
+        } finally {
+          logger debug s"Attempting to close Metric Consumer $mc"
+          mc.close() // Release all associated resources.
+        }
       } catch {
-        case e: Exception => logger.error("Metric consumer failed", e)
-      } finally {
-        logger debug s"Attempting to close Metric Consumer $mc"
-        mc.close() // Release all associated resources.
+        case NonFatal(e) =>
+          logger.error("Could not create a new MetricConsumer for '{}'.", dataDir, e)
+        case e: Throwable =>
+          logger.error("Unrecoverable exception occurred while creating a new MetricConsumer for '{}'. Exiting.",
+            dataDir, e)
+          // If an exception escapes from this method, all future scheduled
```

Comment:
This is some really unfortunate behavior but looks like a credible reason for the zombiesque behavior we've been seeing in Metric Consumer.